app/src/main/java/org/oppia/android/app/devoptions/DeveloperOptionsActivityPresenter.kt

```diff
@@ -58,4 +65,29 @@ class DeveloperOptionsActivityPresenter @Inject constructor(
   fun forceCrash(): Nothing {
     throw RuntimeException("Force crash occurred")
   }
+
+  /** Opens a dialog to force download remote parameters. */
+  fun forceDownloadRemoteParameters() {
+    val dialog = ForceDownloadRemoteParametersDialogFragment.newInstance()
+    dialog.showNow(activity.supportFragmentManager, TAG_FORCE_DOWNLOAD_DIALOG)
+  }
+
+  /** Called when [DeveloperOptionsActivity] is destroyed. */
+  fun handleOnDestroy() {
+    if (restartRequired) {
+      val intent = Intent(activity, SplashActivity::class.java).also {
+        it.action = Intent.ACTION_MAIN
+        it.addCategory(Intent.CATEGORY_LAUNCHER)
+      }
+      activity.startActivity(intent)
+      // App is terminated to ensure a fresh restart and kill all the current process
+      // so that ProcessState can be reinitialised on the fresh restart.
+      exitProcess(0)
+    }
+  }
+
+  /** Marks that a restart is required when the activity is destroyed. */
+  fun markRestartRequired() {
+    restartRequired = true
+  }
```

Comment:
The only usage I can find of this is the`ForceDownloadRemoteParametersDialogFragmentPresenter`, so this function should be private to that presenter.